quantization/nlp/bert/migraphx/README.md

```diff
@@ -0,0 +1,67 @@
+# BERT QDQ Quantization in ONNX for MIGraphX
+There are two main steps for the quantization:
+1. Calibration is done based on SQuAD dataset to get dynamic range of floating point tensors in the model
+2. Q/DQ nodes with dynamic range (scale and zero-point) are inserted to the model
+
+After Quantization is done, you can evaluate the QDQ model by running evaluation script we provide or your own script.
+
+The **e2e_migraphx_bert_example.py** is an end-to-end example for you to reference and run.
+
+## Requirements
+* Please build from latest ONNX Runtime source (see [here](https://onnxruntime.ai/docs/build/eps.html#migraphx)) for now.
+We plan to include TensorRT QDQ support later in ONNX Runtime 1.11 for [ORT Python GPU Package](https://pypi.org/project/onnxruntime-gpu/)
+* MIGraphX 2.8 and above
+* ROCm 5.7 and above (For calibration data)
+* Python 3+
+* numpy 
+* The onnx model used in the script is converted from Hugging Face BERT model. https://huggingface.co/transformers/serialization.html#converting-an-onnx-model-using-the-transformers-onnx-package
+* We use [SQuAD](https://rajpurkar.github.io/SQuAD-explorer/) dataset as default dataset which is included in the repo. If you want to use other dataset to do calibration/evaluation, please either follow the format of squad/dev-1.1.json to add dataset or you have to write your own pre-processing method to parse your dataset.
+
+Some utility functions for dataset processing, data reader and evaluation are from Nvidia TensorRT demo BERT repo,
+https://github.com/NVIDIA/TensorRT/tree/master/demo/BERT
+
+Code TensorRT example has been reused for the MIGraphX Execution Provider to showcase how simple it is to convert over CUDA and TensorRT code into MIGraphX and ROCm within Onnxruntime. Just change the desired Execution Provider and install the proper requirements (ROCm and MIGraphX) and run your script as you did with CUDA.
+
+We've also added a few more input args to the script to help finetune the inference you'd like to run. Feel free to use the --help when running
+
+
+usage: e2e_migraphx_bert_example.py [-h] [--fp16] [--int8] [--model] [--version VERSION] [--batch BATCH] [--seq_len SEQ_LEN] [--doc_stride DOC_STRIDE] [--cal_num CAL_NUM] [--verbose]
+
+options:
+  -h, --help            show this help message and exit
+  --fp16                Perform fp16 quantization on the model before running inference
```

Comment:
Does the model need any head-of-time graph optimizations (like use transformer optimization or converting to packing model to optimize inference)?

Reply:
No that's all done via MIGraphX in this case through the MIGraphX EP that uses the MIGraphX API